modules/core/src/main/scala/notebook/NBSerializer.scala

```diff
@@ -218,44 +234,45 @@ object NBSerializer extends Logging {
 
   implicit val worksheetFormat = Json.format[Worksheet]
 
-  case class Notebook(
-    metadata: Option[Metadata] = None,
-    cells: Option[List[Cell]] = Some(Nil),
-    worksheets: Option[List[Worksheet]] = None,
-    autosaved: Option[List[Worksheet]] = None,
-    nbformat: Option[Int]) {
-    def name = metadata.map(_.name).getOrElse("Anonymous")
-  }
-
   implicit val notebookFormat = Json.format[Notebook]
 
   def fromJson(json: JsValue): Option[Notebook] = {
     logTrace("\r\n**************************************\r\n")
     logTrace(Json.prettyPrint(json))
     logTrace("**************************************\r\n")
     json.validate[Notebook] match {
-      case s: JsSuccess[Notebook] => {
+      case s: JsSuccess[Notebook] =>
         s.get match {
-          case Notebook(None,None,None,None,None) =>
+          case Notebook(None, None, None, None, None) =>
             logWarn("Nothing in the notebook data.")
-            None
-          case notebook =>
-            Some( notebook.cells.map { _ => notebook } getOrElse notebook.copy(cells = Some(Nil)) )
+            throw new NotebookDeserializationError("Got an empty notebook")
```

Comment:
If we want to signal failure, this method should return a `Try[Notebook]` instead of throwing an exception. That way, upper layers can deal with the issue.